### ISSUE TITLE
PathItem deserialization broken because operations field and extensions field are both map-like.

### ISSUE THREAD

**Issue by gcharnock** [juhaku/utoipa#972]
PathItem is made of two `#[serde(flatten)]` "map-like" types. When deserilizing a map, doesn't seem to know which map-like field to put a particular key in, so it puts the key and value in both. This can be reproduced as follows:

```rust
/*
cargo init
cargo add serde
cargo add serde_json
cargo add utoipa


[dependencies]
serde = "1.0.204"
serde_json = "1.0.120"
utoipa = "4.2.3"

*/

use serde_json::json;
use utoipa::openapi::PathItem;

fn main() {
    let json = json! ({
      "post": {
        "responses": {}
      }
    });
    let path_item: PathItem = serde_json::from_value(json).unwrap();
    let serialized = serde_json::to_string_pretty(&path_item).unwrap();
    println!("{}", serialized);
}
```

This prints

```json
{
  "post": {
    "responses": {}
  },
  "post": {
    "responses": {}
  }
}
```

When using a such a schema with Swagger-UI it will produce the error "**Parser error/duplicated mapping**".

I encountered this while trying to merge in custom a custom openapi for a particularly weird endpoint into the rest of the utoipa definition. The workaround for input known to not contain any extensions would be to just to blank the extensions field after deserializing.

```rust
path_item.extensions = None;
```

**Comment by juhaku** [comment]
@gcharnock Good catch, need to fix this for the coming utoipa 5.0.0 version

**Comment by juhaku** [comment]
Related #1062
